%YAML:1.0

map_config:
  has_map: 0
  octree_resolution: 128
  map_path: "/home/chrisliu/ROSws/SV-SLAM_ws/src/SV-SLAM/config/urban27.pcd"


imu_topic: "/camera/imu"
lidar_topic: "/rslidar_points"

sensor_config:
  Tr_lidar2cam: !!opencv-matrix
    rows: 4
    cols: 4
    dt: d
    data: [ -0.00473247314328829,-0.999905181550327,0.0129318060129239,-0.0190059376205962,
            0.0124894465736349,-0.0129900438521302,-0.999837623059367,-0.338465368722495,
            0.999910804733218,-0.00457019359847877,0.0125497373773338,-0.180874466984036,
            0,0,0,1 ]

cam_number: 1

cam_config_0:
  image_topic: "/camera_array/cam0/image_raw"
  enable_segmentor: 1

  #camera calibration
  image_width: 2048
  image_height: 1536

  rotation_rect_matrix: !!opencv-matrix
    rows: 4
    cols: 4
    dt: d
    data: [1,0,0,0,
           0,1,0,0,
           0,0,1,0,
           0,0,0,1]

  # P = K[R|t]
  projection_matrix: !!opencv-matrix
    rows: 3
    cols: 4
    dt: d
    data: [1857.4752797615988, 0., 1039.692658811044,0.0,
           0., 1869.2155909761746, 739.3362262994145,0.0,
           0.0,0.0,1.0,0.0]

  intrinsic_matrix: !!opencv-matrix
    rows: 3
    cols: 3
    dt: d
    data: [1857.4752797615988, 0., 1039.692658811044, 
           0., 1869.2155909761746, 739.3362262994145, 
           0., 0., 1.]

  distortion_coefficients: !!opencv-matrix
    rows: 5
    cols: 1
    dt: d
    data: [-3.0451652061637519e-01, 3.8894492749157322e-02, 1.0517466490567872e-03, -5.5095449411338973e-04, 1.7054732639063463e-01]

segmentor_config:
  show_img: 1
  config_path: "/home/chrisliu/ROSws/SV-SLAM_ws/src/SV-SLAM/lib/mmsegmentation/configs/bisenetv1/bisenetv1_r18-d32_4x4_1024x1024_160k_cityscapes.py"
  checkpoint_path: "/home/chrisliu/ROSws/SV-SLAM_ws/src/SV-SLAM/lib/mmsegmentation/checkpoints/bisenetv1_r18-d32_4x4_1024x1024_160k_cityscapes_20210922_172239-c55e78e2.pth"
  device: "cuda:0"
